pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -912,16 +906,14 @@ protected WriteInEventLoopCallback newObject(Handle<WriteInEventLoopCallback> ha
     }
 
     private void clearPendingMessagesWhenClose() {
```

Comment:
what about changing this name to "closeAndClearPendingMessages" ?

because now the method does something different from the previous behaviour

Reply:
Sure, makes sense to me. Done.